operators/pkg/instancesnapshot-controller/instancesnapshot_controller.go

```diff
@@ -0,0 +1,153 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package instancesnapshot_controller groups the functionalities related to the creation of a persistent VM snapshot.
+package instancesnapshot_controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/utils"
+
+	batch "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	crownlabsv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+)
+
+// ContainersSnapshotOpts contains image names and tags of the containers needed for the VM snapshot.
+type ContainersSnapshotOpts struct {
+	ContainerKaniko    string
+	ContainerImgExport string
+}
+
+// InstanceSnapshotReconciler reconciles a InstanceSnapshot object.
+type InstanceSnapshotReconciler struct {
+	client.Client
+	EventsRecorder     record.EventRecorder
+	Scheme             *runtime.Scheme
+	NamespaceWhitelist metav1.LabelSelector
+	VMRegistry         string
+	RegistrySecretName string
+	ContainersSnapshot ContainersSnapshotOpts
+
+	// This function, if configured, is deferred at the beginning of the Reconcile.
+	// Specifically, it is meant to be set to GinkgoRecover during the tests,
+	// in order to lead to a controlled failure in case the Reconcile panics.
+	ReconcileDeferHook func()
+}
+
+// Reconcile reconciles the status of the InstanceSnapshot resource.
+func (r *InstanceSnapshotReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if r.ReconcileDeferHook != nil {
+		defer r.ReconcileDeferHook()
+	}
+
+	isnap := &crownlabsv1alpha2.InstanceSnapshot{}
+
+	if err := r.Get(ctx, req.NamespacedName, isnap); client.IgnoreNotFound(err) != nil {
+		klog.Errorf("Error when getting InstanceSnapshot %s before starting reconcile -> %s", isnap.Name, err)
+		return ctrl.Result{}, err
+	} else if err != nil {
+		klog.Infof("InstanceSnapshot %s already deleted", isnap.Name)
```

Comment:
Since an error occurred, isnap.Name is empty. You should probably use req.NamespacedName.name